pkg/model/dev.go

```diff
@@ -1313,8 +1316,19 @@ func (b *BuildInfo) Copy() *BuildInfo {
 	args = append(args, b.Args...)
 	result.Args = args
 
+	secrets := BuildSecrets{}
+	for k, v := range b.Secrets {
+		secrets[k] = v
```

Comment:
Instead of passing the original content of the file, let's create a temporal file (and remove it with a defer after the build) with the content of the file after applying env substitutiton.
For example, for `.npmrc`, you would have this file as a secret:
```
egistry=https://registry.npmjs.org/
//***.jfrog.io/artifactory/api/npm/:_authToken=${NPM_TOKEN}
@scope**:registry=https://***.jfrog.io/artifactory/api/npm/private-npm
```
you won't need to commit your token, okteto will replace the token using okteto secrets or your local env vars